src/makefiles/default_rules.mk

```diff
@@ -114,8 +114,30 @@ valgrind: .valgrind
 	rm valgrind.out
 	touch .valgrind
 
+
+#buid up dependency commands
+CC_SRCS=$(wildcard *.cc)
+#check if files exist to run dependency commands on
+ifneq ($(CC_SRCS),)										
+CC_DEP_COMMAND=$(CXX) -M $(CXXFLAGS) $(CC_SRCS)
+endif
+
+ifeq ($(CUDA), true)
+CUDA_SRCS=$(wildcard *.cu)
+#check if files exist to run dependency commands on
+ifneq ($(CUDA_SRCS),)
+NVCC_DEP_COMMAND = $(CUDATKDIR)/bin/nvcc -M $(CUDA_FLAGS) $(CUDA_INCLUDE) -Xcompiler "$(CXXFLAGS)" $(CUDA_SRCS)
```

Comment:
I remember the problem now.

The concern is that this line will become out of sync with this rule:

https://github.com/kaldi-asr/kaldi/blob/5e0e18568f26d0dacd55bdc15b80a69b15a9d232/src/makefiles/cuda_64bit.mk#L9

which _does not_ pass CXXFLAGS to -Xcompiler.

I would suppose that your second -Xcompiler parameter of the command line, since it comes later, could override whatever is in the first -Xcompiler parameter. But this doesn't always work, since you cannot *delete* previously input parameters like include paths, for example.

So that's my concern. Does that make sense?

Reply:
Would the proper fix be to pass CXXFLAGS to -Xcompiler on that line then remove it from the dependency command?  Reguardless since this is purely a dependency extraction I don't think those flags will actually matter.

Reply:
Guys, let me know when you've resolved any issues with this.  I am testing other minor fixes to the build system (e.g. should eliminate the travis failures), see #2947.  We could maybe test the changes together.